File: 30daysOFcode_C#/day 25.cs
using System;
using System.Collections.Generic;
using System.IO;
class Solution {
        static void Main(string[] args)
        {
            int Counter = int.Parse(Console.ReadLine());

            for (int i = 0; i < Counter; i++)
            { 
                int Number= int.Parse(Console.ReadLine());

                if (IsPrime(Number))
                    Console.WriteLine("Prime");
                else
                    Console.WriteLine("Not prime");
            }
        }
        public static bool IsPrime(int n)
        {
            if (n < 2)
                return false;
            for (int i = 2; i * i <= n; i++)
                if (n % i == 0)
                    return false;
            return true;
        }  
}

File: 30daysOFcode_Go/solution_set2/day 25.go
package main

import (
	"fmt"
)

func main() {
	var numbers int
	var number int
	fmt.Scanln(&numbers)
	for i := 0; i < numbers; i++ {
		fmt.Scanln(&number)
		isPrime := true
		if number <= 1 {
			fmt.Println("Not prime")
		} else if number <= 3 {
			fmt.Println("Prime")
		} else if number%3 == 0 || number%2 == 0 {
			fmt.Println("Not prime")
		} else {
			for i := 5; i*i <= number; i = i + 1 {
				if number%i == 0 {
					fmt.Println("Not prime")
					isPrime = false
					break
				}
			}
			if isPrime == true {
				fmt.Println("Prime")
			}
		}
	}
}
